src/VisualStudio/Core/Def/DocumentOutline/ILanguageServiceBrokerShim.cs

```diff
@@ -0,0 +1,20 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Collections.Generic;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.VisualStudio.LanguageServer.Protocol;
+using Microsoft.VisualStudio.Text;
+using Microsoft.VisualStudio.Threading;
+using Newtonsoft.Json.Linq;
+
+namespace Microsoft.VisualStudio.LanguageServices.Implementation.LanguageServiceBrokerShim
+{
+    internal interface ILanguageServiceBrokerShim
+    {
+        Task<JToken?> RequestAsync(ITextBuffer textBuffer, Func<JToken, bool> capabilitiesFilter, string languageServerName, string method, Func<ITextSnapshot, JToken> parameterFactory, CancellationToken cancellationToken);
+    }
```

Comment:
what is this type for?

Reply:
I think @JoeRobich might be the expert here, but my understanding is that we don't want [`Microsoft.VisualStudio.LanguageServices`](https://www.nuget.org/packages/Microsoft.VisualStudio.LanguageServices) to take a dependency on 
 `Microsoft.VisualStudio.LanguageServer.Client.Implementation` because that package is not published to nuget so it would break downstream consumers. Instead, we moved the dependency to `Microsoft.VisualStudio.LanguageServices.Implementation` and exposed those apis via this shim.

Reply:
Yup, more conversation about that at https://github.com/dotnet/roslyn/pull/61784#discussion_r904037872

Reply:
Can we get a comment added to at least explain it or link to a tracking bug or something?

Reply:
Marking as unresolved, but we concluded we didn't need to do this: we have a custom nuspec for this package so the dependencies would have gotten removed.